step-generation/src/utils/pythonFileUtils.ts

```diff
@@ -174,15 +174,25 @@ const _getLidStacks = (
       const { stack } = labwareState[id]
       const nonSlotStackLength = stack.length - 1
       const parentLabware = stack.slice(1, nonSlotStackLength) // excluding stack
-      const isLidStack = parentLabware.every(
+      const deckRiserParentLabware = parentLabware.find(
         parentLabwareId =>
-          allLabwareEntities[parentLabwareId].labwareDefURI === labwareDefURI
+          allLabwareEntities[parentLabwareId].def.parameters.loadName ===
```

Comment:
my Sentry senses are tingling
```suggestion
          allLabwareEntities[parentLabwareId]?.def.parameters.loadName ===
```

Reply:
oh ya good call, thanks!